lib/ReqOpts.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2020 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const clone = require('clone');
+const log = require('./bunyan-api').createLogger('reqOpts');
+
+const ERR_NODATA = 'make sure your data exists in the correct location and is in the expected format.';
+const KIND_MAP = new Map([
+  ['secretKeyRef', 'Secret'],
+  ['secretMapRef', 'Secret'],
+  ['configMapRef', 'ConfigMap'],
+  ['configMapKeyRef', 'ConfigMap']
+]);
+
+module.exports = class ReqOpts {
```

Comment:
instead of duplicating a lot of the code from FetchEnvs, is it possible to just re-use the fetchEnvs component here?